contracts/Registry.sol

```diff
@@ -0,0 +1,40 @@
+pragma solidity ^0.4.18;
+
+
+/**
+ * @title Registry
+ * @dev This contract works as a registry of versions, it holds the implementations for the registered versions.
+ */
+contract Registry {
+  /**
+  * @dev This event will be emitted every time a new implementation is registered
+  * @param version representing the version name of the registered implementation
+  * @param implementation representing the address of the registered implementation
+  */
+  event VersionAdded(string version, address implementation);
+
+
+  // Mapping of versions to implementations of different functions
+  mapping (string => address) private versions;
+
+  /**
+  * @dev Registers a new version with its implementation address
+  * @param version representing the version name of the new implementation to be registered
+  * @param implementation representing the address of the new implementation to be registered
+  */
+  function addVersion(string version, address implementation) public {
```

Comment:
I'm not sure about making this method open to the public. I'd make the Registry `Ownable`, and restrict this to `onlyOwner`.

Reply:
Let's keep it simple for now, we can create `OwnedRegistry` (or different registry governance mechanisms) later. Specifically, for v0.1 we'll need anyone to be able to upload a new version to the registry :)